pkg/vm/stack_item.go

```diff
@@ -72,11 +72,11 @@ func makeStackItem(v interface{}) StackItem {
 		}
 	case []byte:
 		return &ByteArrayItem{
```

Comment:
I have thought about this and decided not to do this, because we push a lot `[]byte` in tests, where is used e.g. as a collection key, so `ByteArray` is expected. Converting to `string` every time is ugly and using `NewByteArray` not differs much from what we have now. So explicit `NewBufferItem` is good IMO.